library/threading.c

```diff
@@ -79,6 +82,50 @@ int (*mbedtls_mutex_unlock)( mbedtls_threading_mutex_t * ) = threading_mutex_unl
  */
 #define MUTEX_INIT  = { PTHREAD_MUTEX_INITIALIZER, 1 }
 
+#else
```

Comment:
I think it's a little confusing and unnecessary to have these functions defined in `threading.c`. I think I'd propose having them in `threading_windows.c` and rename `threading.c` to `threading_pthread.c` to make the code a bit clearer.